src/praisonai-train/praisonai_train/data/protocols.py

```
@@ -0,0 +1,34 @@
+        (disjoint offsets let parallel workers avoid overlap)."""
+        ...
+
+
+@runtime_checkable
+class RowCheck(Protocol):
+    """A single per-row quality check. ``kind`` is 'drop' or 'flag'."""
+
+    name: str
+    kind: str
+
+    def triggered(self, instruction: str, input: str, output: str, cfg: dict) -> bool:
```

Ruff (0.15.21)

[error] Function argument `input` is shadowing a Python builtin

(A002)